### DISCUSSION TITLE
"PermissionError: [Errno 13] Permission denied: '/.cache'" When Building gem5 in Docker

### DISCUSSION THREAD

**Discussion by ghost** [gem5/gem5#941] [Q&A]
I'm trying to build gem5 on an X86 system--running Ubuntu 20.04 using Docker--that I've SSH'ed into from my M1 Mac. On the system, I'm pulling and running the Docker image containing Ubuntu 22.04 and all optional dependencies as provided in the 'building gem5' documentation [page](https://www.gem5.org/documentation/general_docs/building). After running the image and issuing the gem5 build command, I'm met with the error included in the title. The build command I'm using is "scons build/X86/gem5.opt -j 54" with gem5 version 23.1. I get the same error using another one of the Docker images provided on the same page. Is there any way to fix this? I'm not sure what I'm doing wrong.
![Screenshot 2024-03-16 at 4 50 15 PM](https://github.com/gem5/gem5/assets/90482410/ed87d6d6-c705-4f67-8732-927b6220ba0c)

**Comment by ghost** [comment]
Quick follow up! I ended up trying one of the Docker images with "minimum dependencies" and have been successful! No errors occur, but I am met with quite a few warnings. I'm assuming all the warnings are due to the packages missing from this image. Can anyone confirm this? On that note, does anyone know what capabilities these missing packages provide? I'm sorry if this is trivial- I'm new to gem5!
![Screenshot 2024-03-16 at 5 32 36 PM](https://github.com/gem5/gem5/assets/90482410/994f48ab-e295-4b88-b84b-694d7054fa76)

**Reply by robhau** [reply]
Hi,

did you install the dependencies [(Building gem5)](https://www.gem5.org/documentation/general_docs/building)? For example, one dependency is the `libgoogle-perftools-dev` package.
